prow/config/config.go

```diff
@@ -276,6 +278,12 @@ func parseConfig(c *Config) error {
 		if v.MaxConcurrency < 0 {
 			return fmt.Errorf("job %s jas invalid max_concurrency (%d), it needs to be a non-negative number", v.Name, v.MaxConcurrency)
 		}
+		if err := validatePodSpec(v.Name, kube.PresubmitJob, v.Spec); err != nil {
```

Comment:
Can we ensure this has unit test coverage?

In other words I would expect to see something like:
```
cases := []struct {
  name string
  config string // or maybe Config
  err bool
}{
  {
    name: "a config with a job that uses an init container",
    config: `name: foo; spec: initContainers: something`
    err: true,
  },
  // cases for the cross product of job type and validateFoo function failure
}
  for _, tc := range cases { c := yaml.Unmshal(tc.config); actual := parseConfig(&c); actual != tc.expected }
```

Does this already exist somewhere and I'm just missing it?